test/propose.lean

```diff
@@ -0,0 +1,52 @@
+import Std.Data.List.Basic
+import Mathlib.Tactic.Propose
+import Mathlib.Tactic.GuardHypNums
+import Mathlib.Algebra.Associated
+
+theorem foo (L M : List α) (w : L.Disjoint M) (m : a ∈ L) : a ∉ M := fun h => w m h
+
+example (K L M : List α) (w : L.Disjoint M) (m : K ⊆ L) : True := by
+  propose using w
```

Comment:
here and below: assuming that you want these tactics to give deterministic answers, could you add tests using [`Std.Tactic.GuardExpr.guardHyp`](https://leanprover-community.github.io/mathlib4_docs/Std/Tactic/GuardExpr.html#Std.Tactic.GuardExpr.guardHyp) to ensure that `propose` is doing the right thing?